engine/classes/Elgg/Database/EntityTable.php

```diff
@@ -1235,4 +1265,31 @@ function updateLastAction($guid, $posted = null) {
 			return false;
 		}
 	}
-}
\ No newline at end of file
+
+	/**
+	 * Get a user by GUID even if the entity is hidden or disabled
+	 *
+	 * @param int $guid User GUID. Default is logged in user
+	 *
+	 * @return \ElggUser|false
+	 * @throws UserFetchFailureException
+	 * @access private
+	 */
+	public function getUserForPermissionsCheck($guid = 0) {
+		if (!$guid) {
+			return _elgg_services()->session->getLoggedInUser();
```

Comment:
I've been doing this with my own code for some time now for these exact reasons.  I'm starting to dislike the magic default of logged in user more and more as I work around edge cases.